### DISCUSSION TITLE
How to remove car from database

### DISCUSSION THREAD

**Discussion by lytesout2376** [teslamate-org/teslamate#1199] [Q&A] [answered]
I created an instance of teslamate on a raspberry pi and gave it to my dad for his Tesla account. I tested it by signing into my Tesla account to make sure it was working. I was able to sign out (deleted my account from the database) and it allowed him to sign in to his account. But the problem is, because I signed in with my account, my car is showing up on the Dashboards.  There is no info on my car (since I'm signed out) but now he needs to keep hiding the panel for my car.

Is there a map of all of the tables created in the database?  Or better yet, how do I delete my car from his teslamate database?

**Comment by lytesout2376** [comment, marked as answer]
docker-compose exec database psql -U teslamate -d teslamate

SELECT * FROM cars;

DELETE FROM cars WHERE id = [id number here];

**Reply by KingKongKent** [reply]
Hi, when running DELETE FROM cars WHERE id = 1; the Settings page on http://x.x.x.x:4000/settings gets corrupted.
eslamate_1  | Request: GET /settings
teslamate_1  | ** (exit) an exception was raised:
teslamate_1  |     ** (UndefinedFunctionError) function nil.id/0 is undefined
teslamate_1  |         nil.id()
teslamate_1  |         (teslamate 1.21.4) lib/teslamate_web/live/settings_live/index.ex:184: anonymous fn/2 in TeslaMateWeb.SettingsLive.Index.prepare/1
teslamate_1  |         (elixir 1.11.3) lib/enum.ex:2193: Enum."-reduce/3-lists^foldl/2-0-"/3
teslamate_1  |         (teslamate 1.21.4) lib/teslamate_web/live/settings_live/index.ex:16: TeslaMateWeb.SettingsLive.Index.mount/3
teslamate_1  |         (phoenix_live_view 0.15.4) lib/phoenix_live_view/utils.ex:320: anonymous fn/4 in Phoenix.LiveView.Utils.maybe_call_live_view_mount!/4
teslamate_1  |         (telemetry 0.4.2) /opt/app/deps/telemetry/src/telemetry.erl:262: :telemetry.span/3
teslamate_1  |         (phoenix_live_view 0.15.4) lib/phoenix_live_view/static.ex:306: Phoenix.LiveView.Static.call_mount_and_handle_params!/5
teslamate_1  |         (phoenix_live_view 0.15.4) lib/phoenix_live_view/static.ex:131: Phoenix.LiveView.Static.render/3


Returns "Internal Server Error"

**Reply by lytesout2376** [reply]
Was that your only car in the database? If yes, that might be the problem. I ran those commands in a database that had multiple cars and did it on the car that didn’t belong to the Tesla account that was signed in.

**Reply by KingKongKent** [reply]
No, I Sold a Model S, and added a new car a Model X to my Tesla account.  So in my case, i have car 1 and 2, and want to remove car nr 1.
